### ISSUE TITLE
Characters with 10-bit long varicodes are not demodulated properly in Fldigi

### ISSUE THREAD

**Issue by MrMan314** [joshua-jerred/PSK31-Modulation-Generator#2]
Characters with 10-bit long varicodes such as % & ? @ Z ^ { } ~ are not demodulated correctly in Fldigi.  Instead, different characters are printed, or multiple characters are printed.  

Strangely, this does not happen to the single quote (') however.  

I have tried in all modes and the same seems to happen

**Comment by joshua-jerred** [comment]
I just tested it and have had the same results as described. Thank you for bringing this up as I had not noticed it at all.

I'm working on a fix right now!

**Comment by joshua-jerred** [comment]
edb74f4f8e145ad217179e24de2fb7fd07b13dba
Looks like it was a small logic issue that only supported 9 bit varicode when the length was greater than 8 bits. I tested it with BPSK and it seems to be working with all characters now. Let me know if this fixes it for you.

I don't plan on improving this repository anymore as it has been merged into my larger modulation library [MWAV](https://github.com/joshua-jerred/MWAV) which might be useful depending on what you're doing.
